hawkbit-rest/hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootController.java

```diff
@@ -492,20 +458,57 @@ private Status handleClosedCase(final DdiActionFeedback feedback, final String c
             @PathVariable("actionId") @NotEmpty final Long actionId) {
         LOG.debug("provideCancelActionFeedback for target [{}]: {}", controllerId, feedback);
 
-        final Target target = controllerManagement.getByControllerId(controllerId)
-                .orElseThrow(() -> new EntityNotFoundException(Target.class, controllerId));
-
+        final Target target = findTargetWithExceptionIfNotFound(controllerId);
         final Action action = findActionWithExceptionIfNotFound(actionId);
-        if (!action.getTarget().getId().equals(target.getId())) {
-            LOG.warn(GIVEN_ACTION_IS_NOT_ASSIGNED_TO_GIVEN_TARGET, action.getId(), target.getId());
-            return ResponseEntity.notFound().build();
-        }
+        verifyActionAssignedToTarget(target, action);
 
         controllerManagement
                 .addCancelActionStatus(generateActionCancelStatus(feedback, target, actionId, entityFactory));
         return ResponseEntity.ok().build();
     }
 
+    @Override
+    public ResponseEntity<DdiDeploymentBase> getControllerInstalledAction(@PathVariable("tenant") final String tenant,
+            @PathVariable("controllerId") final String controllerId, @PathVariable("actionId") final Long actionId,
+            @RequestParam(value = "actionHistory", defaultValue = DdiRestConstants.NO_ACTION_HISTORY) final Integer actionHistoryMessageCount) {
+        LOG.debug("getControllerInstalledAction({})", controllerId);
+
+        final Target target = findTargetWithExceptionIfNotFound(controllerId);
+        final Action action = findActionWithExceptionIfNotFound(actionId);
+        verifyActionAssignedToTarget(target, action);
+
+        if (!action.isActive() && !action.isCancelingOrCanceled()) {
```

Comment:
Minor thing but I would suggest to reverse the condition. So move the notFound() return into the condition block and the expected successful path currently inside the block out and to the end, usually easier to read that way.

Reply:
Done. Inverted the if condition